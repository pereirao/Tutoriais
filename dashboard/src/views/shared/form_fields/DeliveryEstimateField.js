// @flow

import React, {PureComponent} from 'react';
import {Dropdown} from 'semantic-ui-react';
import {connect} from 'react-redux';
import * as Ent from '@minibar/store-business/src/utils/ent';
import {order_helpers} from '../../../business/order';
import style_sheet from './form_fields.module.css';

const {deliveryEstimateOptions, makeDeliveryEstimateParam} = order_helpers;

type DeliveryEstimateFieldProps = { onChangeText: (string, Object) => void, options: Array<Object>, formFields: { [string]: MBFormField } };

class DeliveryEstimateField extends PureComponent {
  props: DeliveryEstimateFieldProps

  static defaultProps = {name: 'delivery_estimate'}
  input: ReactClass<*>

  render() {
    const disableDropdown = this.props.formFields.use_delivery_service.value === 'true';

    return (
      <DeliveryEstimateDropdown
        disabled={disableDropdown}
        order_id={this.props.order_id}
        onChangeText={this.props.onChangeText}
        className={style_sheet.textInput}/>
    );
  }
}

const DeliveryEstimateDropdownSTP = () => {
  const findOrder = Ent.find('order');
  return (state, {order_id}) => {
    const order = findOrder(state, order_id);
    return {
      order,
      options: deliveryEstimateOptions(order)
    };
  };
};
const connectDeliveryEstimateDropdown = connect(DeliveryEstimateDropdownSTP);
const DeliveryEstimateDropdown = connectDeliveryEstimateDropdown(({
                                                                    options,
                                                                    order,
                                                                    onChangeText,
                                                                    ...dropdown_props
                                                                  }) => {
  const handleChange = (_e, {value}) => {
    onChangeText(makeDeliveryEstimateParam(order, value), true);
  };
  return (
    <Dropdown selection options={options} onChange={handleChange} {...dropdown_props} />
  );
});
export default DeliveryEstimateField;
