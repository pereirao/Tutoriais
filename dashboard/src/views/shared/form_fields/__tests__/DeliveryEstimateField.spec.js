import React from 'react';

import DeliveryEstimateField from '../DeliveryEstimateField';

it('renders correctly', () => {
  expect(shallow(<DeliveryEstimateField />)).toMatchSnapshot();
});
