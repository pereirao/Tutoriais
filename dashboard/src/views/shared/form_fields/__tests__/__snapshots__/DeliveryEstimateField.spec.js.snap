// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Connect(Component)
  className="textInput"
/>
`;
